safekeeper/src/handler.rs

```diff
@@ -150,9 +164,17 @@ impl postgres_backend::Handler for SafekeeperPostgresHandler {
         self.ttid = TenantTimelineId::new(tenant_id, timeline_id);
 
         match cmd {
-            SafekeeperPostgresCommand::StartWalPush => ReceiveWalConn::new(pgb).run(self),
+            SafekeeperPostgresCommand::StartWalPush => match ReceiveWalConn::new(pgb).run(self) {
```

Comment:
Why we can't use ? here as before?

Reply:
There hast to be the same type `        match cmd {` returns, so we do `.context` on it later.
Two lower branches return `anyhow::Result` still.

Reply:
Ah, I see. Lack of context is a bit sad here, though we still have tid to find ends. It would be nice to log error earlier while ttid span is still there or pass it somehow...

Reply:
I have added logging around these lines, but not sure how else I can carry the context around.